hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/TinyLfuBlockCache.java

```diff
@@ -188,21 +196,58 @@ public void cacheBlock(BlockCacheKey cacheKey, Cacheable value, boolean inMemory
 
   @Override
   public void cacheBlock(BlockCacheKey key, Cacheable value) {
+    cacheBlockUtil(key, value, false);
+  }
+
+  private void cacheBlockUtil(BlockCacheKey key, Cacheable value, boolean deepClonedOnHeap) {
```

Comment:
These can be in cacheBlock() only as per previous comment.

Reply:
Done